extension/json/json_common.cpp

```diff
@@ -31,24 +32,65 @@ string ThrowPathError(const char *ptr, const char *end, const bool binder) {
 	}
 }
 
-static inline idx_t ReadString(const char *ptr, const char *const end, const bool escaped) {
+struct JSONKeyReadResult {
+public:
+	static inline JSONKeyReadResult Empty() {
+		return {idx_t(0), string()};
+	}
+
+	static inline JSONKeyReadResult WildCard() {
+		return {1, "*"};
+	}
+
+	inline bool IsValid() {
+		return chars_read != 0;
+	}
+
+	inline bool IsWildCard() {
+		return key == "*";
+	}
+
+public:
+	idx_t chars_read;
+	string key;
+};
+
+static inline JSONKeyReadResult ReadString(const char *ptr, const char *const end, const bool escaped) {
 	const char *const before = ptr;
 	if (escaped) {
+		auto key = make_unsafe_uniq_array<char>(end - ptr);
+		idx_t key_len = 0;
+
+		bool backslash = false;
 		while (ptr != end) {
 			if (*ptr == '"') {
-				break;
+				if (!backslash) {
+					break;
+				}
+				backslash = false;
+			} else if (*ptr == '\\') {
+				if (!backslash) {
+					backslash = true;
```

Comment:
Backslash seems to only be set to false on a quote or backslash - what if the string is e.g. `\x"`?

Reply:
Hm, but then returning an empty result here doesn't seem correct to me - shouldn't we match `x\` as the string literal `'x\'` in that case, instead of returning an empty result? Could we add such a test?